test/service/test_query_shortener.py

```diff
@@ -34,7 +34,8 @@ def lambda_name(cls) -> str:
         return 'service'
 
     def _shorten_query_url(self, url, expect_status=None):
-        response = requests.post(self.base_url + '/url', json={'url': url})
+        response = requests.post(self.base_url.set(path='/url').url,
+                                 json={'url': url})
```

Comment:
This fits on one line (barely); personally I think it looks better as you've written it but you may want to unwrap